src/ensembl/src/content/app/browser/drawer/drawerReducer.ts

```diff
@@ -8,16 +8,22 @@ export default function drawer(
   action: ActionType<typeof drawerActions>
 ): DrawerState {
   switch (action.type) {
-    case getType(drawerActions.changeDrawerView):
+    case getType(drawerActions.changeDrawerViewForGenome):
       return {
         ...state,
-        drawerView: action.payload
+        drawerView: { ...state.drawerView, ...action.payload }
       };
-    case getType(drawerActions.toggleDrawer):
+    case getType(drawerActions.toggleDrawerForGenome):
+      const genomeId = Object.keys(action.payload)[0];
+      let drawerOpenedValue = Object.values(action.payload)[0];
+      let drawerViewForGenome = drawerOpenedValue
+        ? state.drawerView[genomeId]
+        : '';
```

Comment:
Very confused about what’s going on here:

1) If the payload structure of type `{ [genomeId: string]: boolean }` is not convenient enough, because you have to get the keys of this object (which is only one key) and all values of this object (which is only one value), then perhaps the payload of type `{ genomeId: string, isOpen: boolean }` will be better suited for the task?

2) don’t understand what’s going on with drawerViewForGenome

Reply:
I have refactored this part of the code in 494761c. What I was trying to do here was to check if the drawer is closed and if so to reset the drawer view.

Reply:
Now it's so much cleaner 👍